SEED-platform/seed
# refactor flilteredReadings to exclude cols and rows

#### Any background context you want to provide?
When viewing an individual property's meters, one can choose to filter by meter type (or scenario). There is a dropdown menu where a user can select or unselect meter types. Below the dropdown is a table of all meters. Their column headers tend to have a timestamp and a meter type entry (ex: 'Start Time', 'End Time', 'Electricity...', 'Natural Gas...'). Unselecting a meter type will only remove the meter type column, not the row. This is unhelpful as rows with no meter data are still visible.

#### What's this PR do?
Updates the filterByMeterLabels function to remove rows that do not contain the selected meters. The function would return a row if it contained a "timeColumn" header: `timeColumns = ['start_time', 'end_time', 'month', 'year'];`. Since all meter entries contian a `start_time` or `end_time`, all rows would be returned, but without the meter readings. 

The new function will only return a row if it contains the selected meter label. Columns are filtered in a different area and have remained unchanged.


#### How should this be manually tested?
Inventory > Property with multiple meter types > meters > unselect a meter type > the table rows should only reflect meters that contain a selected meter type

#### What are the relevant tickets?
#3159

#### Screenshots (if appropriate)
**OLD:**
One Meter Selected: Showing rows that have empty values for selected meter type.
![Screen Shot 2022-03-09 at 10 00 36 AM](https://user-images.githubusercontent.com/58446472/157492251-5136310c-b3f4-47f5-96ef-4be480a9f618.png)


No Meters Selected: Showing all meters, but no meter data
![Screen Shot 2022-03-09 at 10 00 48 AM](https://user-images.githubusercontent.com/58446472/157492272-9b75cb8f-7693-43bc-8730-5e6f8304b7fb.png)


**NEW:**
All Meters Selected:
![Screen Shot 2022-03-09 at 8 34 37 AM](https://user-images.githubusercontent.com/58446472/157492792-06eff375-70a3-4a03-84de-c5450f763416.png)


One Meter Selected: Rows without selection removed
![Screen Shot 2022-03-09 at 8 34 25 AM](https://user-images.githubusercontent.com/58446472/157492759-e656562e-f1ed-4b0c-8f44-40eab6d91c5a.png)


No Meters Selected: All rows removed
![Screen Shot 2022-03-09 at 8 33 39 AM](https://user-images.githubusercontent.com/58446472/157492698-2ac31650-3df3-4fb1-a5d0-d4d6a7807cbd.png)
